darshan-util/pydarshan/darshan/experimental/plots/plot_dxt_heatmap.py

```diff
@@ -0,0 +1,412 @@
+    if initial_yticklabels.size < n_ylabels:
+        # if there are less tick marks available than requested,
+        # use the original tick mark labels
+        y_ticklabels = initial_yticklabels
+
+    else:
+        # use the original tick marks to make a new array that
+        # contains a subset of the original tick mark labels
+        tick_idx = np.round(
+            np.linspace(0, initial_yticklabels.size - 1, n_ylabels)
+        ).astype(int)
+        y_ticklabels = initial_yticklabels[tick_idx]
+    return y_ticklabels
+
+
+def set_x_axis_ticks_and_labels(
+    jointgrid: Any, agg_df: pd.DataFrame, n_xlabels: int = 4
+):
+    """
+    Sets the x-axis tick mark locations and labels.
+
+    Parameters
+    ----------
+
+    jointgrid: a ``sns.axisgrid.JointGrid`` object.
+
+    agg_df: a ``pd.DataFrame`` containing the aggregated data determined
+    by the input modules and operations.
+
+    n_xlabels: the number of x-axis tick marks to create. Default is 4.
+
+    """
+    # retrieve the x-axis tick mark locations and labels
+    xticks = get_x_axis_ticks(ax=jointgrid.ax_joint, n_xlabels=n_xlabels)
+    xticklabels = get_x_axis_tick_labels(agg_df=agg_df, n_xlabels=n_xlabels)
+    # set the x-axis ticks and labels
+    jointgrid.ax_joint.set_xticks(xticks)
+    jointgrid.ax_joint.set_xticklabels(xticklabels, minor=False)
+
+
+def set_y_axis_ticks_and_labels(jointgrid: Any, n_ylabels: int = 6):
+    """
+    Sets the y-axis tick mark locations and labels.
+
+    Parameters
+    ----------
+    jointgrid: a ``sns.axisgrid.JointGrid`` object.
+
+    n_ylabels: The number of y-axis tick mark labels to create. Default is 6.
+
+    """
+    # retrieve the y-axis tick mark locations and labels
+    yticks = get_y_axis_ticks(ax=jointgrid.ax_joint, n_ylabels=n_ylabels)
+    yticklabels = get_y_axis_tick_labels(ax=jointgrid.ax_joint, n_ylabels=n_ylabels)
+    # set the new y-axis tick locations and labels
+    jointgrid.ax_joint.set_yticks(yticks)
+    jointgrid.ax_joint.set_yticklabels(yticklabels, minor=False)
+
+
+def remove_marginal_graph_ticks_and_labels(marg_x: Any, marg_y: Any):
+    """
+    Removes the frame, tick marks, and tick mark
+    labels for the marginal bar graphs.
+
+    Parameters
+    ----------
+    marg_x : a x-axis marginal bar graph object.
+
+    marg_y : a y-axis marginal bar graph object.
+    """
+    # turn the frame off for both bar graphs
+    marg_x.axis("off")
+    marg_y.axis("off")
+    # remove all tick mark labels for both bar graphs
+    marg_x.tick_params(
+        axis="x", bottom=False, labelbottom=False, top=False, labeltop=False
+    )
+    marg_y.tick_params(
+        axis="y", left=False, labelleft=False, right=False, labelright=False
+    )
+    marg_x.tick_params(
+        axis="y", left=False, labelleft=False, right=False, labelright=False
+    )
+    marg_y.tick_params(
+        axis="x", bottom=False, labelbottom=False, top=False, labeltop=False
+    )
```

Comment:
Can reduce duplication/maintenance burden here:

```diff
+    for marg in [marg_x, marg_y]:
+        # turn the frame off for both bar graphs
+        marg.axis("off")
+        # remove all tick mark labels for both bar graphs
+        marg.tick_params(
+            axis="x", bottom=False, labelbottom=False, top=False, labeltop=False
+        )
+        marg.tick_params(
+            axis="y", left=False, labelleft=False, right=False, labelright=False
+        )
```

Reply:
The new version uses this simplification (and passes tests), so I'll resolve this.